src/Nethermind/Nethermind.JsonRpc/Modules/BlockFinderExtensions.cs

```diff
@@ -75,8 +77,47 @@ public static SearchResult<Block> SearchForBlock(this IBlockFinder blockFinder,
             }
 
             return block == null && !allowNulls
-                ? new SearchResult<Block>($"{blockParameter.BlockHash?.ToString() ?? blockParameter.BlockNumber?.ToString() ?? blockParameter.Type.ToString()} could not be found", ErrorCodes.ResourceNotFound)
+                ? new SearchResult<Block>($"Block {blockParameter.BlockHash?.ToString() ?? blockParameter.BlockNumber?.ToString() ?? blockParameter.Type.ToString()} could not be found", ErrorCodes.ResourceNotFound)
                 : new SearchResult<Block>(block);
         }
+
+        public static IEnumerable<SearchResult<Block>> SearchForBlocksOnMainChain(this IBlockFinder blockFinder, BlockParameter fromBlock, BlockParameter toBlock)
```

Comment:
We do have RequireCanonical which probably makes this redundant?

Reply:
No, because it is search for blocks. I don't want search for single block, but for range of blocks

Reply:
Why aren't we going block by block producing results until we run out of blocks or our trace limit?